cmd/event-notifier.go

```diff
@@ -230,7 +230,14 @@ func (en *eventNotifier) SendListenerEvent(arn string, event []NotificationEvent
 
 	ch, ok := en.internal.connectedListeners[arn]
 	if ok {
-		ch <- event
+		// To avoid blocking of send event over the channel
+		// while inside critical section.
+		//FIXME: The intermediary channel can be replaced by
+		//the actual http.ResponseWriter.
+		select {
+		case ch <- event:
+		default:
```

Comment:
Can we log an error here?

Reply:
@harshavardhana What would the error message be? What do we intend to inform our users here?

Reply:
```
errorIf(errors.New("blocked channel"), "Unable to send the event to %s, possibly no receiver", arn) 
```